casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -155,16 +155,18 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
         toBondSeq(bondsPar.head)
       }
 
-  private def bondsQuerySource(name: String = "__SCALA__"): String =
+  private def rholangCall(uri: String, args: Any*): String =
     s"""
-       | new rl(`rho:registry:lookup`), SystemInstancesCh, posCh in {
-       |   rl!(`rho:rchain:systemInstancesRegistry`, *SystemInstancesCh) |
-       |   for(@(_, SystemInstancesRegistry) <- SystemInstancesCh) {
-       |     @SystemInstancesRegistry!("lookup", "pos", *posCh) |
-       |     for(pos <- posCh){ pos!("getBonds", "$name") }
+       | new rl(`rho:registry:lookup`), objCh in {
+       |   rl!(`$uri`, *objCh) |
+       |   for(@(_, obj) <- objCh) {
+       |     @obj!(${args.map(_.toString).mkString(",")})
```

Comment:
Please revert this, this obscures what rholang is being executed and prevents plugging in debugging. I'm surprised to be saying this, but not all deduplication makes sense.

Reply:
yeah, I agree. It was an experiment I did and I forgot to undo